### PR TITLE
Cleanup docs

### PR DESCRIPTION
**Pull request by nalimilan**
Ensure that all exported methods are included in the manual, so that we can enable strict=true, which ensures that doctests pass.
Update doctests to latest Julia version.

### CODE REVIEW

File: src/weights.jl [hunk 1]
```diff
@@ -536,8 +536,9 @@ wsumtype(::Type{T}, ::Type{W}) where {T,W} = typeof(zero(T) * zero(W) + zero(T)
 wsumtype(::Type{T}, ::Type{T}) where {T<:BlasReal} = T
 
 """
-    wsum!(R, A, w, dim; init=true)
-
+    wsum!(R::AbstractArray, A::AbstractArray,
+          w::AbstractWeights{<:Real}, dim::Int;
+          init=true)
```
Comment (bkamins) [inline src/weights.jl:541]:
```suggestion
          init::Bool=true)
```

File: src/weights.jl [hunk 2]
```diff
@@ -562,9 +563,24 @@ end
 
 ## extended sum! and wsum
 
+"""
+    sum!(R::AbstractArray, A::AbstractArray,
+         w::AbstractWeights{<:Real}, dim::Int;
+         init=true)
```
Comment (bkamins) [inline src/weights.jl:569]:
```suggestion
         init::Bool=true)
```

File: src/weights.jl [hunk 3]
```diff
@@ -562,9 +563,24 @@ end
...
+
+Compute the weighted sum of `A` with weights `w` over the dimension `dim` and store
+the result in `R`. If `init=false`, the sum is added to `R` rather than starting
+from zero.
+"""
 Base.sum!(R::AbstractArray, A::AbstractArray, w::AbstractWeights{<:Real}, dim::Int; init::Bool=true) =
     wsum!(R, A, w, dim; init=init)
 
+"""
+    sum(v::AbstractArray, w::AbstractVector; [dims])
```
Comment (bkamins) [inline src/weights.jl:579]:
```suggestion
    sum(v::AbstractArray, w::AbstractVector{<:Real}; [dims])
```

**Review by bkamins — APPROVED** [review]
Looks good. I have left some minor comments. Thank you!

**Comment by nalimilan** [comment]
Thanks!

**Pull request merged**